vkontakte/content/contents/code/vkontakte.js

```diff
@@ -1,398 +1,339 @@
-                    else{                    
-                        if (debugMode == true) { Tomahawk.log("ERROR!: Unknown error while searching track"); }
-                        callback({error:'Unknown error while searching track'})
+        Tomahawk.log("Sending " + count + " of queued resolve requests");
+        var code = 'return [' + queries.join(',') + '];';        
+        Tomahawk.log("Prepared 'execute' code:" + code);
+        that._apiCall('execute', {code:code}).then(function(results) {
+            for (var result in results.response) {
+                result = results.response[result];
+                var tracks = [];
+                tracks = tracks.concat.apply(tracks, result.result.map(function(item) {return item.items;}));
+                //Leave unique only
+                var u = {}, a = [];
+                Tomahawk.log(JSON.stringify(tracks));
+                for(var i = 0, l = tracks.length; i < l; ++i){
+                    if(u.hasOwnProperty(tracks[i].url)) {
+                        continue;
                     }
+                    a.push(tracks[i]);
+                    u[tracks[i].url] = 1;
+                }
+                var _qid = JSON.parse(decodeURIComponent(result.qid));
+                tracks = a.map(that._convertTrack, 
+                    {
+                        artist: _qid[0],
+                        album : _qid[1],
+                        title : _qid[2],
+                    });
+                for(var r in saved_queue[decodeURIComponent(result.qid)]) {
+                    saved_queue[decodeURIComponent(result.qid)][r](tracks);
                 }
-            } catch(err) {
-                Tomahawk.log("Error while parsing response");
             }
-        })
+        });
     },
 
+    resolve: function (artist, album, title) {
+        var that = this;
 
-    /**
-        VK#search(artist, song, callback)
-        - artist (String): Artist
-        - song (String): Song
-        - callback (Function): Function to be called when search compete, to obtain results. 
-    **/    
-    search: function(artist, song, duration, callback){
-        if (debugMode == true) { Tomahawk.log("Seaching: "+artist+" - "+song); }
+        var qid = JSON.stringify([artist, album, title]);
 
-        if(this.search_method == undefined){
-            this.determineSearchMethod(function(response){
-                if (debugMode == true) { Tomahawk.log("Search method: "+response.search_method); }
+        var promise = new Promise(function (resolve, reject) {
+            if(!(qid in that._queue)) {
+                that._queue[qid] = [];
+            }
+            Tomahawk.log(resolve);
+            that._queue[qid].push(resolve);
+            Tomahawk.log(JSON.stringify(that._queue));
+        });
 
-                VK.search_method = response.search_method
+        if (!(this._batching)) {
+            setTimeout(function(){ that._batchResolve(that); }, 1000);
```

Comment:
Have you checked what timeout tomahawk has for waiting on results from a resolver?